src/Toolbox/Util/Graph/BFS.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE BangPatterns #-}
+
+-- | Breadth-First-Search (BFS)
+module Toolbox.Util.Graph.BFS
+  ( -- * internal state for BFS
+    BFSState (..),
+    BFSPath (..),
+    MultiBFSPath (..),
+
+    -- * single-seed use case
+    stepBFS,
+    runStagedBFS,
+
+    -- * multi-seed use case
+    stepMultiseedBFS,
+    runMultiseedStagedBFS,
+  )
+where
+
+import Control.Monad (join)
+import Control.Monad.Extra (loopM)
+-- import Control.Monad.IO.Class (liftIO)
```

Comment:
i guess we can get rid of this line?

Reply:
Oh. sorry. This is deleted in the next PRs but revived here after rebasing. Thank you for the catch.